### ISSUE TITLE
How to choose m rays and some questions about eq(5)

### ISSUE THREAD

**Issue by AAAeray** [YifYang993/CR-NeRF-PyTorch#1]
Hi YifYang,
Thanks for your great work, I've just read your paper and don't read your code yet, I don't understand some questions:
1. How to choose m rays during training, choose m pixels in a single image randomly, or m neighbors of a pixel? or some else rule? A set of m rays' features match m pixels' colors in loss function or a single pixel's color?
2. In eq(5),why there need the second term (the beta term). And why it can be modeld as the D_theta5 network.
Thanks a lot.

**Comment by YifYang993** [comment]
Hi AAAeray,
1.   Please refer to our code datasets/phototourism_mask_grid_sample.py line 243 to line 266. We uniformly sample the m rays, and the m rays match the m corresponding pixel colors.
2.   Please refer to our recently submitted version in arxiv, the beta term is for ensuring that the content of the transformed image (D_{\theta_4}(\mT_{\theta_3}(\mF^{\mathrm{cr}}))) is the same of the image (D_{\theta_4}(\mF^{\mathrm{cr}}))  whose appearance is not transformed. In other words, we want to keep the content unchanged after modifying the appearance.
